src/Tutor.Core/LearnerModel/DomainOverlay/KcMasteryService.cs

```diff
@@ -49,23 +51,31 @@ public Result<List<InstructionalItem>> GetInstructionalItems(int knowledgeCompon
             var kcMastery = _kcMasteryRepository.GetFullKcMastery(knowledgeComponentId, learnerId);
             if (kcMastery == null) return Result.Fail("Learner not enrolled in KC: " + knowledgeComponentId);
 
-            var result = kcMastery.SelectInstructionalItems();
+            var result = kcMastery.RecordInstructionalItemSelection();
```

Comment:
The name implies that this is a command when it is more of a query with a side-effect that produces an event...
Working with aggregates is hard :)